newrelic/hooks/mlmodel_openai.py

```diff
@@ -133,61 +145,119 @@ def wrap_chat_completion_create(wrapped, instance, args, kwargs):
+
+    # Get trace information
+    available_metadata = get_trace_linking_metadata()
+    span_id = available_metadata.get("span.id", "")
+    trace_id = available_metadata.get("trace.id", "")
+
+    # Get conversation ID off of the transaction
+    custom_attrs_dict = transaction._custom_params
+    conversation_id = custom_attrs_dict.get("conversation_id", "")
+
+    settings = transaction.settings if transaction.settings is not None else global_settings()
+    app_name = settings.app_name
+    completion_id = str(uuid.uuid4())
+
     ft_name = callable_name(wrapped)
     with FunctionTrace(ft_name) as ft:
         try:
             response = wrapped(*args, **kwargs)
         except Exception as exc:
-            error_attributes = openai_error_attributes(exc, kwargs)
-            exc._nr_message = error_attributes.pop("error.message")
+            exc_organization = getattr(exc, "organization", "")
+
+            notice_error_attributes = {
+                "http.statusCode": getattr(exc, "http_status", ""),
+                "error.message": getattr(exc, "_message", ""),
+                "error.code": getattr(getattr(exc, "error", ""), "code", ""),
+                "error.param": getattr(exc, "param", ""),
+                "completion_id": completion_id,
+            }
+            exc._nr_message = notice_error_attributes.pop("error.message")
             ft.notice_error(
-                attributes=error_attributes,
+                attributes=notice_error_attributes,
             )
+            # Gather attributes to add to embedding summary event in error context
+            error_chat_completion_dict = {
+                "id": completion_id,
+                "appName": app_name,
+                "conversation_id": conversation_id,
+                "api_key_last_four_digits": api_key_last_four_digits,
+                "span_id": span_id,
+                "trace_id": trace_id,
+                "transaction_id": transaction.guid,
+                "response.number_of_messages": len(request_message_list),
+                "request.model": kwargs.get("model") or kwargs.get("engine") or "",
+                "request.temperature": kwargs.get("temperature", ""),
+                "request.max_tokens": kwargs.get("max_tokens", ""),
+                "vendor": "openAI",
+                "ingest_source": "Python",
+                "response.organization": "" if exc_organization is None else exc_organization,
+                "duration": ft.duration,
+                "error": True,
+            }
+            transaction.record_custom_event("LlmChatCompletionSummary", error_chat_completion_dict)
+
+            error_response_id = str(uuid.uuid4())
+
+            create_chat_completion_message_event(
+                transaction,
+                app_name,
+                request_message_list,
+                completion_id,
+                span_id,
+                trace_id,
+                "",
+                error_response_id,
```

Comment:
We already have code inside create_message to default this correctly if it doesn't exist so just let that code run, otherwise we end up with uuid-0 instead of just uuid.
```suggestion
                None,
```